pyartcd/pyartcd/pipelines/build_microshift.py

```diff
@@ -2,28 +2,33 @@
 import logging
 import os
 import re
+import io
 import traceback
 import click
+import copy
 from collections import namedtuple
 from pathlib import Path
 from typing import Dict, Iterable, List, Optional, Tuple
 
 from artcommonlib.arch_util import brew_arch_for_go_arch
 from artcommonlib.assembly import AssemblyTypes
-from artcommonlib.util import get_ocp_version_from_group, new_roundtrip_yaml_handler
+from artcommonlib.util import get_ocp_version_from_group, new_roundtrip_yaml_handler, get_assembly_release_date
 from artcommonlib import exectools
 from doozerlib.util import isolate_nightly_name_components
-from ghapi.all import GhApi
```

Comment:
Why replace ghapi?

Reply:
I refactor that create pr function to reduce git clone and use a more concise way to create pr